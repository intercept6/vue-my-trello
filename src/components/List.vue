<template>
<div class="list">
  <div class="list-header">
    <p class="list-title">{{title}}</p>
    <p class="list-counter">total: {{totalCardInList}}</p>
    <div class="delete-list" @click="removeList">x</div>
  </div>
  <draggable group="cards"
             :list="cards"
             @end="$emit('change')"
  >
    <Card v-for="(item, index) in cards"
          :body="item.body"
          :key="item.id"
          :cardIndex="index"
          :listIndex="listIndex"
    ></Card>
  </draggable>
  <cardAdd :listIndex="listIndex"></cardAdd>
</div>
</template>

<script>
import Card from './Card';
import CardAdd from './CardAdd';
import draggable from 'vuedraggable';
import {computed} from '@vue/composition-api';

export default {
  name: "List",
  props: {
    title: {
      type: String,
      required: true,
    },
    cards: {
      type: Array,
      required: true,
    },
    listIndex: {
      type: Number,
      required: true,
    },
  },
  setup(props, context) {
    const totalCardInList = computed(() => {
      return props.cards.length;
    });

    const $store = context.root.$store;
    const removeList = () => {
      if (confirm('本当にこのリストを削除しますか？')) {
        $store.dispatch('removeList', {listIndex: props.listIndex});
      }
    };

    return {
      totalCardInList,
      removeList,
    };
  },
  // computed: {
  //   totalCardInList() {
  //     return this.cards.length;
  //   }
  // },
  // methods: {
  //   removeList() {
  //     if (confirm('本当にこのリストを削除しますか？')) {
  //       this.$store.dispatch('removeList', {listIndex: this.listIndex});
  //     }
  //   },
  // },
  components: {
    Card,
    CardAdd,
    draggable,
  },
};
</script>
