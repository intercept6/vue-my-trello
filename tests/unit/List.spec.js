import {mount, shallowMount} from '@vue/test-utils';
import List from '../../src/components/List';

describe('List.vue', () => {
  test('スナップショットテスト', () => {
    const wrapper = shallowMount(List, {
      propsData: {

        title: 'test title',
        cards: [],
        listIndex: 0,
      },
    });

    expect(wrapper.element).toMatchSnapshot();
  });


  test('描画テスト', () => {
    const wrapper_shallow = shallowMount(List, {
      propsData: {

        title: 'test title',
        cards: [],
        listIndex: 0,
      },
    });

    console.log(wrapper_shallow.html());

    const wrapper = mount(List, {
      propsData: {

        title: 'test title',
        cards: [],
        listIndex: 0,
      },
    });

    console.log(wrapper.html());
  });
});
