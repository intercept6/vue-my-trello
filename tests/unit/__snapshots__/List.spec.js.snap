// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List.vue スナップショットテスト 1`] = `
<div
  class="list"
>
  <div
    class="list-header"
  >
    <p
      class="list-title"
    >
      test title
    </p>
     
    <p
      class="list-counter"
    >
      total: 0
    </p>
     
    <div
      class="delete-list"
    >
      x
    </div>
  </div>
   
  <draggable-stub
    clone="function(t){return t}"
    element="div"
    group="cards"
    list=""
  />
   
  <cardadd-stub
    listindex="0"
  />
</div>
`;
